BestFloww/BestFlow
# Left/Right Buttons for IntentLister

## Pull Request Summary
Adds a Left and Right button to Analysis page that switches between sets of intents to display. Can also use arrow keys on keyboard to activate the buttons. Buttons get disabled when reaching their respective boundary (index is 0 for left, index is within 3 of the end for right).

Several files were changed to achieve this. A notable other change was that BaseButton now accepts a label prop in case there is no text, since it was otherwise impossible for testing to get that button (and also it's good for a11y). But for the left/right buttons, most of the work is done in IntentLister on top of the existing index-related code.

The increment/skip left/right icons are placeholders - we can prob work to find pairs that look more alike.

### TODO:
- [x] Figure out why it doesn't respond to arrow keys, let it accept A/D as well
- [x] Update arrow icons to be thicker
- [x] Simultaneous key press
- [x] Validation for having aria label if no text in BaseButton

### Change types


### Testing
I don't think I can have testing for the keyboard presses by Thursday, but the increment/decrement methods themselves (as well as the buttons) are tested!

Edit: the skip to front/end have not been tested. Will do testing for that in another PR over the next week.

### How to manually test this PR (QA instructions)
Go to Analysis Page. Use the left/right arrow buttons to switch between intents. Use A/D keys and left/right arrow keys to do the same thing. Hold CTRL while doing either to skip to the extreme in that direction. Notice that the icons change as CTRL is held and released.

## Questions

Please check the code for how key press events are handled! I feel it's ugly but I don't have enough experience to know if/where it can be trimmed down.
I followed a combination of these posts:
https://stackoverflow.com/questions/37440408/how-to-detect-esc-key-press-in-react-and-how-to-handle-it
https://stackoverflow.com/questions/5597060/detecting-arrow-key-presses-in-javascript

Please see this post for how to handle multi-key shortcuts:
https://www.gavsblog.com/blog/detect-single-and-multiple-keypress-events-javascript

Please see this post for conditional prop requirement added to BaseButton:
https://stackoverflow.com/questions/42299335/react-props-set-isrequired-on-a-prop-if-another-prop-is-null-empty
